adapters/repos/db/crud.go

```diff
@@ -264,7 +268,13 @@ func (db *DB) Merge(ctx context.Context, merge objects.MergeDocument,
 
 	err := idx.mergeObject(ctx, merge, repl, tenant)
 	if err != nil {
-		return errors.Wrapf(err, "merge into index %s", idx.ID())
+		wrapped := errors.Wrapf(err, "merge into index %s", idx.ID())
+		switch err.(type) {
```

Comment:
please use `errors.As(err, &objects.ErrDirtyWriteOfDeletedObject{})` instead of switch, bc switch will not work for wrapped errors